src/ethereum/base_types.py

```diff
@@ -636,4 +637,69 @@ def to_signed(self) -> int:
 Bytes20 = Bytes
 Bytes32 = Bytes
 Bytes64 = Bytes
-BytesMutable256 = bytearray
+Bytes256 = Bytes
+
+
+def _setattr_function(self: Any, attr: str, value: Any) -> None:
+    if self._frozen:
+        raise Exception("Mutating frozen dataclasses is not allowed.")
+    else:
+        object.__setattr__(self, attr, value)
+
+
+def _delattr_function(self: Any, attr: str) -> None:
+    if self._frozen:
+        raise Exception("Mutating frozen dataclasses is not allowed.")
+    else:
+        object.__delattr__(self, attr)
+
+
+def _make_init_function(f: Callable) -> Callable:
+    def init_function(self: Any, *args: Any, **kwargs: Any) -> None:
+        will_be_frozen = kwargs.pop("_frozen", True)
+        object.__setattr__(self, "_frozen", False)
+        f(self, *args, **kwargs)
+        self._frozen = will_be_frozen
+
+    return init_function
+
+
+def slotted_freezable(cls: Any) -> Any:
+    """
+    Monkey patches a dataclass so it can be frozen by setting `_frozen` to
+    `True` and uses `__slots__` for efficiency.
+
+    Instances will be created frozen by default unless you pass `frozen=False`
```

Comment:
It [looks](https://github.com/ethereum/eth1.0-specs/pull/275/files#diff-b8a22b1b8cce59cc5b6d01201a7f2d75394c510f127101f09a93e69a3d679a87R659) like the argument is named `_frozen`? Not 100% sure though.

Reply:
Well spotted.